tests/unit/spec/base/core-datalayer.js

```diff
@@ -25,6 +25,60 @@ describe('core-datalayer.js', function() {
         });
     });
 
+    describe('updateDataLayerPush', function() {
+
+        beforeAll(function() {
+            Mozilla.Analytics.updateDataLayerPush();
+        });
+
+        beforeEach(function() {
+           var link = '<a id="link" href="https://www.mozilla.org/en-US/firefox/new/">';
+            $(link).appendTo('body');
+            window.linkElement = $('#link')[0];
+        });
+
+        afterEach(function() {
+            $('body').remove('#link');
```

Comment:
We should clean up after each test/suite. Say we have a test later that expects `dataLayer.length` to be a certain value. These (unrelated) tests would cause unexpected results.